### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by juan-castillo-wolox**
## Summary

* create users controller 
* add new errors 
* add contants helper 
* add getValidations middleware 
* add new route users to sign-up 
* create encryption service 
* create users service
* Install http-status-codes dependency

## Known Issues

None

## Trello Card

[Link to the associated Trello card.](https://trello.com/c/Y57syO8h/35-sign-up)

### CODE REVIEW

File: app/controllers/users.controller.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const httpStatusCodes = require('http-status-codes');
+const { hashPassword } = require('../services/encryptions');
```
Comment (jonnyalexbh) [inline app/controllers/users.controller.js]: I think these kinds of methods could be in a helper, what do you think?

File: app/controllers/users.controller.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+exports.signUp = (req, res, next) => {
+  const { firstName, lastName, email, password } = req.body;
+  return hashPassword({ firstName, lastName, email, password })
+    .then(userService.createUser)
+    .then(user => {
+      logger.info(`The user ${user.firstName} has been created`);
+      res.status(httpStatusCodes.CREATED).send({ user_id: user.id });
+    })
+    .catch(error => {
+      logger.error(error.message);
```
Comment (jonnyalexbh) [inline app/controllers/users.controller.js]: unnecessary logger here

File: app/helpers/constants.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Regex
+exports.EMAIL = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/gim;
+exports.WOLOX_DOMAIN = 'wolox.co';
```
Comment (jonnyalexbh) [inline app/helpers/constants.js]: What happens if the user has an email like this: `test@wolox.com.ar` or `test.wolox.com.mx`?

File: app/helpers/constants.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+// Messages
+exports.MISSING = 'This field is required';
+exports.INVALID_CONTENT = 'The content of this field is invalid';
+exports.INVALID_EMAIL = 'This field must be a valid email';
+exports.INVALID_EMAIN_DOMAIN = `The email must include the ${this.WOLOX_DOMAIN} domain`;
+exports.INVALID_PASSWORD_LENGTH = 'must be a minimum of 8 characters';
+exports.INVALID_ALPHANUMERIC = 'This field must be alphanumeric';
+
+exports.SALT_ROUNDS = 10;
```
Comment (jonnyalexbh) [inline app/helpers/constants.js]: this should be env var

File: app/helpers/constants.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+// Regex
```
Comment (jonnyalexbh) [inline app/helpers/constants.js]: I put the `constant` file in a folder called `constant/index.js`

File: app/controllers/users.controller.js (hunk 1 above)
Comment (jonnyalexbh) [reply app/controllers/users.controller.js]: Also you think if the user exists what error are you returning ?

Comment (juan-castillo-wolox) [reply app/controllers/users.controller.js]: yes, It could be

File: app/controllers/users.controller.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+const httpStatusCodes = require('http-status-codes');
+const { hashPassword } = require('../helpers/encryptions');
+
```
Comment (jonnyalexbh) [inline app/controllers/users.controller.js:3]: Where are you valid if the user exists?

Comment (juan-castillo-wolox) [reply app/controllers/users.controller.js:3]: It is directly validated with the sequilize schema, however in the next commit where validation is requested in the tests, I create an error handler to specify the error

File: app/helpers/constants.js (hunk 3 above)
Comment (juan-castillo-wolox) [reply app/helpers/constants.js]: This was change by a regex validation where is validated that the email contain  wolox.<something else>

File: app/controllers/users.controller.js (hunk 6 above)
Comment (jonnyalexbh) [reply app/controllers/users.controller.js:3]: great

**Comment by jonnyalexbh** [comment]
Good Job! 🥇

**Review by jonnyalexbh — APPROVED** [review]

**Pull request merged**